arangod/RocksDBEngine/RocksDBIndex.cpp

```diff
@@ -248,12 +248,22 @@ Result RocksDBIndex::update(transaction::Methods& trx, RocksDBMethods* mthd,
   // It is illegal to call this method on the primary index
   // RocksDBPrimaryIndex must override this method accordingly
   TRI_ASSERT(type() != TRI_IDX_TYPE_PRIMARY_INDEX);
+  // only if the insert needs to see the changes of the update, enable indexing:
+  bool tempEnableIndexing = mthd->isIndexingDisbled() && hasExpansion() && unique();
+  if (tempEnableIndexing) {
+    mthd->EnableIndexing();
+  }
 
   Result res = remove(trx, mthd, oldDocumentId, oldDoc, mode);
   if (!res.ok()) {
```

Comment:
mmh. Indexing is not turned off here.
Why not create a struct `IndexingEnabler` with RAII semantics as I suggested to you? That would have been clearer.